src/components/Articles.jsx

```diff
@@ -0,0 +1,159 @@
+
+/**
+ * @class Articles
+ * @extends {React.Component}
+ */
+class Articles extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      articles: [],
+      loading: true,
+      sortBy: [],
+    };
+    this.sortArticles = this.sortArticles.bind(this);
+    this.fetchArticles = this.fetchArticles.bind(this);
+    this.SortArticleBy = [];
+  }
+
+  /**
+   * @method componentDidMount - Runs after the page has been rendered
+   * @return {void}
+   * Retrieves article sort value by calling getSortOptions()
+   * and assigns it to instance variable
+   * Makes a call to get list of Articles
+   * Listens for a change in the ArticleStore
+   */
+  componentDidMount() {
+    this.SortArticleBy = this.getSortOptions()[0];
+    ArticleStore.on('change', this.fetchArticles);
+    SiteActions.fetchArticles(this.props.match.params.source, this.SortArticleBy);
+  }
+
+  /**
+   * @method componentWillReceiveProps - Runs after page has been rendered and props have changed
+   * @return {void}
+   * Retrieves article sort value by calling getSortOptions()
+   * and assigns it to instance variable
+   * Makes a call to get list of updated Articles
+   * Sets the state with new
+   */
+  componentWillReceiveProps(props) {
+    this.SortArticleBy = this.getSortOptions()[0];
+    SiteActions.fetchArticles(props.match.params.source, this.SortArticleBy);
+    this.setState({
+      loading: true,
+      sortBy: this.getSortOptions(this.props),
+      articles: ArticleStore.getArticles(),
+    });
+  }
+
+  /**
+   * @method componentWillUnmount - Runs before component is removed from the DOM
+   * @return {void}
+   * Removes the change Listener from ArticleStore
+   */
+  componentWillUnmount() {
+    ArticleStore.removeListener('change', this.fetchArticles);
+  }
+
+  /**
+   * @method getSortOptions - Listens for an event and makes an action call to the Api
+   * @return {array} - returns an array of sort by options
+   */
+  getSortOptions() {
+    const { match } = this.props;
+    const { params } = match;
+    return params.sort.split(',');
+  }
+
+  /**
+   * @method fetchArticles - Sets the state of articles to
+   * data retrieved from ArticleStore; loading to false;
+   * and sortBy to an array of options available for sorting
+   * @return {void}
+   */
+  fetchArticles() {
+    this.setState({
+      articles: ArticleStore.getArticles() || [],
+      loading: false,
+      sortBy: this.getSortOptions(this.props),
+    });
+  }
+
+  /**
+   * @method sortArticles - Listens for an event and makes a call to get list of
+   * articles from a particular source and sort option
+   * @param {event} event - Accepts an event parameter
+   * @return {void}
+   */
+  sortArticles(event) {
+    SiteActions.fetchArticles(this.props.match.params.source, event.target.value);
+  }
+
+  render() {
+    // Declare newsArticles variable and assign the current state to it
+    const newsArticles = this.state;
```

Comment:
Don't put any logic inside the render method. Can you extract this atleast to it's own compnent. This lets you avoid confusion when it comes to updating your or changing your components